### PR TITLE
Fix Airplay metadata, auto Airplay port selection, fix Airplay device names

### PR DESCRIPTION
**Pull request by cmfcmf**
I recommend to review this PR commit by commit. Here is a description of each commit:
- https://github.com/badaix/snapcast/commit/05575f46e605a3aa490f7ca1ed7c0960a9b73ca7: Fixes the code to automatically select a port for shairport-sync if the given port is already in use. 
  - shairport-sync sends a different message since 3.0 when the port cannot be opened (https://github.com/mikebrady/shairport-sync/commit/945483d9c34eaa39b5911fa1501a5741583dee11)
  - to send logs to stderr, the `--use-stderr` parameter is needed since 3.3 (https://github.com/mikebrady/shairport-sync/blob/master/RELEASENOTES.md#version-33)
- https://github.com/badaix/snapcast/commit/31a9e30666f3cbb5b3b4664414e182d3129906af: Before, all instances of shairport-sync would output metadata to the same pipe. This is now fixed by including the port in the pipename.
- https://github.com/badaix/snapcast/commit/4fe4310ff2cedfa8e1ae7def96006da36624ab1e: Add expat to cmake and build docs
- https://github.com/badaix/snapcast/pull/537/commits/0aed04f9b3b21cc65e8da980b4eca54dfa7172a6: Fix reading from metadata pipe
  - Reading failed with an EOF error until the first metadata is written. Fixed by retrying on EOF errors.
- https://github.com/badaix/snapcast/pull/537/commits/c6cbc24163b17ccbbaec46d53055b8901265a18e: Corrects the argument escaping of the `--name` parameter. Up until now, the Airplay devices all included `"` in their display names (i.e.,they showed up as `"Kitchen Speaker"` instead of `Kitchen Speaker` on iOS devices).

### CODE REVIEW

**Comment by cmfcmf** [comment]
I have removed the commit that parses cover art from this PR since it causes some issue for me. I'll open another PR for that feature if I find out why. This PR now only contains fixes.

File: server/streamreader/airplay_stream.cpp [hunk 1]
```diff
@@ -150,12 +142,17 @@ void AirplayStream::pipeReadLine()
     {
         try
         {
-            int fd = open(pipePath_.c_str(), O_RDONLY | O_NONBLOCK);
+            // We cannot open the file with O_NONBLOCK, because we need to wait for the pipe to be ready before proceeding.
+            // We therefore set the O_NONBLOCK flag only after the pipe has been successfully opened.
+            int fd = open(pipePath_.c_str(), O_RDONLY);
+            int flags = fcntl(fd, F_GETFL, 0);
+            fcntl(fd, F_SETFL, flags | O_NONBLOCK);
```
Comment (cmfcmf) [inline server/streamreader/airplay_stream.cpp]: Something must be wrong with this change. `snapserver` hang indefinitely when trying to stop it using `sudo service snapserver stop`

Comment (cmfcmf) [reply server/streamreader/airplay_stream.cpp]: I was able to fix this issue and updated the description above.

**Comment by badaix** [comment]
Thanks!

**Pull request merged**